src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -8541,6 +8541,126 @@ public void RefFieldsConsideredManaged_Indirect()
             Assert.True(comp.GetTypeByMetadataName("StructWithIndirectRefField").IsManagedTypeNoUseSiteDiagnostics);
         }
 
+        [WorkItem(62618, "https://github.com/dotnet/roslyn/issues/62618")]
+        [Theory]
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void RefAssignValueScopeMismatch_01(LanguageVersion languageVersion)
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void Main()
+    {
+        Span<int> s = new[] { 1 };
+        M(ref s);
+    }
+    static void M(ref Span<int> s)
+    {
+        Span<int> local = stackalloc int[] { 1 };
+        ref Span<int> rL = ref local;
+        rL = ref s; // 1
+        rL = local;
+    }
+}";
+            var comp = CreateCompilationWithSpanAndMemoryExtensions(source, parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion));
+            comp.VerifyEmitDiagnostics(
+                // (13,9): error CS8374: Cannot ref-assign 's' to 'rL' because 's' has a narrower escape scope than 'rL'.
```

Comment:
Is the error message right? In this case, I thought `s` had a wider escape scope than `rL`. The error message should say "different" or swap the positions.